dali/python/nvidia/dali/plugin/mxnet.py

```diff
@@ -128,109 +131,122 @@ class DALIGenericIterator(_DALIMXNetIteratorBase):
+    last_batch_policy = DROP, last_batch_padded = False  -> last batch = ``[5, 6]``, next iteration will return ``[2, 3]``
     """
     def __init__(self,
                  pipelines,
                  output_map,
                  size=-1,
                  reader_name=None,
                  data_layout='NCHW',
-                 fill_last_batch=True,
+                 fill_last_batch=None,
                  auto_reset=False,
                  squeeze_labels=True,
                  dynamic_shape=False,
-                 last_batch_padded=False):
+                 last_batch_padded=False,
+                 last_batch_policy=LastBatchPolicy.FILL):
+
+        # check the assert first as _DaliBaseIterator would run the prefetch
+        self._output_names_map = [x[0] for x in output_map]
+        self._output_categories_map = [x[1] for x in output_map]
+        self._output_categories = {DALIGenericIterator.DATA_TAG, DALIGenericIterator.LABEL_TAG}
+        assert set(self._output_categories_map) <= self._output_categories, \
+            "Only DATA_TAG and LABEL_TAG are allowed"
+        assert len(set(self._output_names_map)) == len(self._output_names_map), \
+            "output_names in output_map should be distinct"
+        self.output_map = output_map
+
         super(DALIGenericIterator, self).__init__(
             pipelines,
             size,
             reader_name,
             fill_last_batch,
             last_batch_padded,
-            auto_reset)
+            auto_reset,
+            last_batch_policy)
         self._squeeze_labels = squeeze_labels
         self._dynamic_shape = dynamic_shape
         # Use double-buffering of data batches
         self._data_batches = [[None] for i in range(self._num_gpus)]
         self._current_data_batch = 0
-        self._output_names_map = [x[0] for x in output_map]
-        self._output_categories_map = [x[1] for x in output_map]
-        self._output_categories = {DALIGenericIterator.DATA_TAG, DALIGenericIterator.LABEL_TAG}
-        assert set(self._output_categories_map) <= self._output_categories, \
-            "Only DATA_TAG and LABEL_TAG are allowed"
-        assert len(set(self._output_names_map)) == len(self._output_names_map), \
-            "output_names in output_map should be distinct"
-        self.output_map = output_map
 
-        # We need data about the batches (like shape information),
-        # so we need to run a single batch as part of setup to get that info
-        for p in self._pipes:
-            with p._check_api_type_scope(types.PipelineAPIType.ITERATOR):
-                p.schedule_run()
         self._first_batch = None
-        self._first_batch = self.next()
+        try:
+            self._first_batch = self.next()
+        except StopIteration:
+            assert False, "It seems that there is not data in the pipeline, please check if last_batch_policy is set PARTIAL and batch size is bigger than the shard size"
```

Comment:
```suggestion
            assert False, "It seems that there is no data in the pipeline. This may happen if `last_batch_policy` is set to PARTIAL and the requested batch size is greater than the shard size."
```

Reply:
Done